From 73c0c45e9275169f9207cfafb79c63d8c431341c Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Tue, 11 Jan 2022 17:12:35 +0200
Subject: [PATCH 03/25] Revert "s32gen1: Add fdt_pcie_spis_fixup in env"

Cleanup fixup that is not strictly necessary. Linux MSIs are now enabled
by default and if one still wants to use the IRQ mechanism supported by
the PCIe AXI Bridge, they can simply remove the "msi-parent" property
from the Linux device-tree PCIe node.

This reverts commit 449f682deca3d0fe9d3130d83d0807468ad13e53.
Issue: ALB-7437

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32.h      | 13 -------------
 include/configs/s32g274a.h |  7 -------
 include/configs/s32r45.h   |  6 ------
 3 files changed, 26 deletions(-)

diff --git a/include/configs/s32.h b/include/configs/s32.h
index 08f0eb1a87..c34f3b393f 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -279,18 +279,6 @@
 #define PFE_INIT_CMD ""
 #endif
 
-#ifdef CONFIG_PCIE_S32GEN1
-#define PCIE_SET_MSI_CONTROLLER \
-	"fdt_pcie_set_gic=fdt addr ${fdt_addr}; fdt resize; " \
-		"fdt get value gic_phandle " \
-		"/interrupt-controller@50800000 phandle; " \
-		"fdt set /pcie@\\${pcie_addr} msi-parent <\\${gic_phandle}>;\0"
-#endif
-
-#ifndef PCIE_MSIS_ENV_SETTINGS
-#define PCIE_MSIS_ENV_SETTINGS	""
-#endif
-
 #if !defined(PCIE_EXTRA_ENV_SETTINGS)
 #if defined(CONFIG_PCIE_S32GEN1) || defined(CONFIG_FSL_PFENG)
 #define PCIE_EXTRA_ENV_SETTINGS \
@@ -448,7 +436,6 @@
 	GMAC_EXTRA_ENV_SETTINGS \
 	PFE_EXTRA_ENV_SETTINGS \
 	PCIE_EXTRA_ENV_SETTINGS \
-	PCIE_MSIS_ENV_SETTINGS \
 
 #undef CONFIG_BOOTCOMMAND
 
diff --git a/include/configs/s32g274a.h b/include/configs/s32g274a.h
index 7d0104c034..c771a0d155 100644
--- a/include/configs/s32g274a.h
+++ b/include/configs/s32g274a.h
@@ -10,13 +10,6 @@
 #ifndef __S32G274A_H
 #define __S32G274A_H
 
-#ifdef CONFIG_PCIE_S32GEN1
-#define PCIE_MSIS_ENV_SETTINGS	\
-	PCIE_SET_MSI_CONTROLLER \
-	"fdt_pcie0_spis_fixup=pcie_addr=40400000; run fdt_pcie_set_gic; \0" \
-	"fdt_pcie1_spis_fixup=pcie_addr=44100000; run fdt_pcie_set_gic; \0"
-#endif
-
 #include <configs/s32-gen1.h>
 
 #if defined(CONFIG_TARGET_S32G2XXAEVB)
diff --git a/include/configs/s32r45.h b/include/configs/s32r45.h
index 4e958b888a..502712a3a3 100644
--- a/include/configs/s32r45.h
+++ b/include/configs/s32r45.h
@@ -10,12 +10,6 @@
 #ifndef __S32R45_H
 #define __S32R45_H
 
-#ifdef CONFIG_PCIE_S32GEN1
-#define PCIE_MSIS_ENV_SETTINGS	\
-	PCIE_SET_MSI_CONTROLLER \
-	"fdt_pcie0_spis_fixup=pcie_addr=40400000; run fdt_pcie_set_gic; \0"
-#endif
-
 #include <configs/s32-gen1.h>
 
 #if defined(CONFIG_TARGET_S32R45EVB)
-- 
2.17.1

